Fix loading saved kb

MarcCote opened the pull request:
Since `State` now takes a `KnowledgeBase` object as its first argument, when we reload a game we should use the KB that was saved along with it.

EDIT: This PR also avoid using `TimeLimit.max_episode_seconds` which was deprecated recently in OpenAI's Gym (see https://github.com/openai/gym/pull/1402).

EDIT2: This PR also avoid using `trials` which was removed from EnvSpec (see https://github.com/openai/gym/commit/c7bae99a06861004cf17e792e5ef88de00198e8d#diff-3dff15d44236dd4f0f823a6ddb7e8c9b).

tavianator commented on textworld/generator/world.py:
```diff
@@ -196,7 +196,7 @@ def state(self) -> State:
     @state.setter
     def state(self, state: State) -> None:
-        self._state = State(KnowledgeBase.default().logic)
+        self._state = State(state._logic)  # or should it be `state.copy()`?
```
That would be better.  It would need to be this I think
```python
self._state = state.copy()
self._update()
```

tavianator commented on textworld/logic/tests/test_query.py:
```diff
@@ -0,0 +1,59 @@
+def test_find_predicates_in_scope():
+    state = _make_state()
+    facts = query.find_predicates_in_scope(state)
+    assert set(facts) == set([
```
You can write `set` literals like `{a, b, ...}`

MarcCote replied:
Oh! that wasn't supposed to be in this PR :package:

tavianator commented on textworld/generator/game.py:
```diff
@@ -447,6 +447,9 @@ def deserialize(cls, data: Mapping) -> "Game":
             game.main_quest = Quest.deserialize(data["main_quest"])
 
+        # Replace default KB with the one saved with the game.
+        game.world._state._logic = game.kb.logic
```
I think we should add `GameLogic` as a parameter to `State.deserialize()` rather than mess with private fields.

MarcCote replied:
Ideally, we wouldn't need a State object in world. We only need a set of propositions, actually.

MarcCote commented:
It's still missing a unit test but I removed the state variable from the World class. That should be cleaner.

MarcCote commented:
@tavianator I changed my approach about how to reload existing KB saved along with a game. Let me know what you think (second commit).

I also fixed some other deprecation in Gym (first commit).
